contracts/ERC20Safe.sol

```diff
@@ -205,19 +248,132 @@ contract ERC20Safe is Initializable, BridgeRole, Pausable {
      *        0x + endpoint_name_length (4 bytes) + endpoint_name + gas_limit (8 bytes) +
      *        00 (ArgumentsPresentProtocolMarker)
      */
-    function depositWithSCExecution(address tokenAddress, uint256 amount, bytes32 recipientAddress, bytes calldata callData) public whenNotPaused {
+    function depositWithSCExecution(
+        address tokenAddress,
+        uint256 amount,
+        bytes32 recipientAddress,
+        bytes memory callData
+    ) public whenNotPaused {
         uint112 batchNonce;
         uint112 depositNonce;
-        (batchNonce, depositNonce) = _deposit_common(tokenAddress, amount, recipientAddress);
-        emit ERC20SCDeposit(batchNonce, depositNonce, callData);
+        (batchNonce, depositNonce) = _deposit_common(tokenAddress, amount, recipientAddress, callData);
     }
 
-
-    function _deposit_common(address tokenAddress, uint256 amount, bytes32 recipientAddress) internal returns (uint112 batchNonce, uint112 depositNonce) {
+    function _deposit_common(
+        address tokenAddress,
+        uint256 amount,
+        bytes32 recipientAddress,
+        bytes memory callData
+    ) internal returns (uint112 batchNonce, uint112 depositNonce) {
         require(whitelistedTokens[tokenAddress], "Unsupported token");
         require(amount >= tokenMinLimits[tokenAddress], "Tried to deposit an amount below the minimum specified limit");
         require(amount <= tokenMaxLimits[tokenAddress], "Tried to deposit an amount above the maximum specified limit");
 
+        _processDelayedTransactions(tokenAddress);
+
+        uint256 currentBlock = block.number;
```

Comment:
Do you think it will be better to work with `block.timestamp` instead of `block.number`, or, at least keep also the timestamp when the deposit was made just to simplify the logic of time computing when we decide we can execute a transaction?